### PR TITLE
Food proficiencies 1

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary

SUMMARY: Content "Adds basic food proficiencies"

#### Purpose of change
Food recipes need proficienices.

#### Describe the solution
Posting this as a draft for feedback, because I am not sure how many of these to do. I know a lot of cooking and could get deep into this rabbithole.

- Knife skills: This represents specific ability at rapid chef knife work. It should almost never impact failure rate, but having it will speed up a lot of recipes by a bit.
- Baking: basic, bread, and desserts. This set seems pretty uncontroversial.
- ~~Cooking: Meat or Veggies. These represent basic ability to take a core ingredient and make it taste good. One could argue this is represented by the cooking skill (although I am considering the possibility of just removing the skill), but I also would like them to potentially step off as prerequisites into more advanced proficiencies. Thoughts?~~ I think I have decided to not go with these.
- Regional cooking proficiencies - currently Asian and European. These could be split, or could have sub-proficiencies, for specific areas. Japanese and Chinese food are pretty different, as are German and Italian. I am not sure if these merit proficiencies or not but they seem logical to me.
- Rendering fats and oils: Adding this because it's got a bit of technical skill to it.

#### Describe alternatives you've considered
Setting this up as a draft to discuss alternatives.

#### Testing
Draft
#### Additional context
This is extremely rough now, don't bother correcting typos

### CODE REVIEW

**Comment by LaVeyanFiend** [comment]
This may be a bit too specific, but I would argue that [blank]-free cooking would be a proficiency, involving attempting to match the substitute to the characteristics of whatever  the normal ingredient is(ex. texturizing pulse or soy protein into meat substitutes).

**Comment by I-am-Erk** [comment]
We'll have to see, I think mostly those recipes are represented as a single entry that allows substitutes which would make it hard.

**Comment by Fosheze** [comment]
I love this. Just a random thought though. What about adding a concentration proficiency for things like making maple syrup, distilling alcohol, making sugar out of various things, etc. Basically anything that requires boiling one material down to get another without ruining it. The main one that comes to mind would be the maple syrup because I have helped make that and knowing how to boil down the sap without burning the hell out of it is definitely a skill. This could also, if possible, be a cross over proficiency for both cooking and chemistry.

**Comment by I-am-Erk** [comment]
Concentration is an idea, it would come up in chemistry as well and is fairly similar between the two. For stuff like maple syrup, confectionery might be another good choice.

**Comment by Night-Pryanik** [comment]
I don't like regional cooking proficiencies. Even if Japanese cuisine differs from Chinese cuisine and they both differs from German or Chilean cuisines, the _methods_ of cooking are basically the same. One either fries or boils or bakes the food, no matter the basic food material.

Also I don't like "cooking: meat or veggies" exactly because of the reason you described in OP - this proficiency is perfectly governed by the cooking skill itself and thus is excessive.

**Comment by I-am-Erk** [comment]
"methods of cooking" aren't the only thing represented by a proficiency. The regional proficiencies would represent the ability to more effectively prepare a specialized dish, or substitute ingredients to still create something that tastes pretty good. I'm open to alternative suggestions of course, but overall I do think there needs to be some way to express further more specific/complex recipes.

**Comment by Night-Pryanik** [comment]
> ability to more effectively prepare a specialized dish

Do you think a high-class chef who all their life cooked food solely from Italian cuisine will fail at their first time cooking food from Japanese cuisine? Or will make, say, sushi 900% time more slower simply because they never cooked it before? Cooking rice shouldn't suddenly become much more harder because they are cooking it for sushi and not for risotto as they used to.

**Comment by I-am-Erk** [comment]
No, neither I nor the proficiency system thinks that. However, the proficiency system matches my own experience, which is that being very skilled at making sushi doesn't mean that you're equally good at making risotto, nor vice versa.

File: data/json/proficiencies/food.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
+[
+  {
+    "type": "proficiency",
+    "id": "prof_asian_food",
+    "name": { "str": "Cooking Asian Food" },
```
Comment (Rail-Runner) [inline data/json/proficiencies/food.json:5]: Maybe it'd be better to name those "Asian Cuisine" and "European Cuisine", respectively? Sounds better to me. And perhaps description in English, too.

File: data/json/proficiencies/food.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    "id": "prof_deep_frying",
+    "name": { "str": "Deep Frying" },
+    "description": "You know how to manage deep fried food perfectly without modern amenities.",
+    "can_learn": true,
+    "time_to_learn": "4 h"
+  },
+  {
+    "type": "proficiency",
+    "id": "prof_knifeskills",
+    "name": { "str": "Knife Skills" },
```
Comment (Rail-Runner) [inline data/json/proficiencies/food.json:69]:
```suggestion
    "name": { "str": "Food Slicing" },
```
Just to make it clear that this has nothing to do with combat.

Comment (I-am-Erk) [reply data/json/proficiencies/food.json:69]: Irl it is called "knife skills" pretty specifically, but yeah we might want to call it "culinary knife skills"

File: data/json/proficiencies/food.json (hunk 1 above)
Comment (I-am-Erk) [reply data/json/proficiencies/food.json:5]: I considered that too, I'm down with it.

**Comment by I-am-Erk** [comment]
Closing this as I've decided to hold off until after stable to do any more proficiencies. Thanks for the feedback, all.

I will be removing the cooking meat and veggies proficiencies, and will probably keep and expand the regional cuisine ones. Stay tuned!

**Pull request closed without merging**